<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>788</width>
    <height>600</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>171</width>
      <height>581</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="hlMain">
     <item>
      <widget class="QGroupBox" name="vbControls_2">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <layout class="QVBoxLayout" name="vbControls">
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <item>
         <widget class="QGroupBox" name="gbRotation">
          <property name="layoutDirection">
           <enum>Qt::LeftToRight</enum>
          </property>
          <layout class="QVBoxLayout" name="gbRoation">
           <item>
            <layout class="QFormLayout" name="Rotations">
             <property name="fieldGrowthPolicy">
              <enum>QFormLayout::ExpandingFieldsGrow</enum>
             </property>
             <property name="rowWrapPolicy">
              <enum>QFormLayout::DontWrapRows</enum>
             </property>
             <property name="labelAlignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
             </property>
             <item row="4" column="0">
              <widget class="QLabel" name="labelA">
               <property name="enabled">
                <bool>true</bool>
               </property>
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="layoutDirection">
                <enum>Qt::LeftToRight</enum>
               </property>
               <property name="autoFillBackground">
                <bool>false</bool>
               </property>
               <property name="lineWidth">
                <number>1</number>
               </property>
               <property name="text">
                <string>Ring A</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
               </property>
              </widget>
             </item>
             <item row="7" column="0">
              <widget class="QLabel" name="labelB">
               <property name="text">
                <string>Ring B</string>
               </property>
              </widget>
             </item>
             <item row="10" column="0">
              <widget class="QLabel" name="labelC">
               <property name="text">
                <string>Ring C</string>
               </property>
              </widget>
             </item>
             <item row="4" column="1">
              <widget class="QSlider" name="hsRotationA">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>0</height>
                </size>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="7" column="1">
              <widget class="QSlider" name="hsRotationB">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="10" column="1">
              <widget class="QSlider" name="hsRotationC">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="0" column="0" colspan="2">
              <widget class="QCheckBox" name="cbIsAnimated">
               <property name="layoutDirection">
                <enum>Qt::RightToLeft</enum>
               </property>
               <property name="text">
                <string>Animated</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0" colspan="2">
              <widget class="QCheckBox" name="cbFixedCam">
               <property name="layoutDirection">
                <enum>Qt::RightToLeft</enum>
               </property>
               <property name="text">
                <string>fixed Camera</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="gbProjection_2">
          <layout class="QVBoxLayout" name="gbProjection">
           <item>
            <widget class="QRadioButton" name="rbPerspective">
             <property name="text">
              <string>Perspective</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="rbOrthogonal">
             <property name="text">
              <string>Orthogonal</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="gbClipping_2">
          <layout class="QVBoxLayout" name="gbClipping">
           <item>
            <layout class="QFormLayout" name="Clipping">
             <item row="1" column="0">
              <widget class="QLabel" name="labelFar">
               <property name="text">
                <string>Far</string>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="labelNear">
               <property name="text">
                <string>Near</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QDoubleSpinBox" name="dsbNear">
               <property name="decimals">
                <number>2</number>
               </property>
               <property name="maximum">
                <double>100.000000000000000</double>
               </property>
               <property name="stepType">
                <enum>QAbstractSpinBox::DefaultStepType</enum>
               </property>
               <property name="value">
                <double>1.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QDoubleSpinBox" name="dsbFar">
               <property name="maximum">
                <double>100.000000000000000</double>
               </property>
               <property name="value">
                <double>90.000000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="Sliders">
          <item>
           <layout class="QVBoxLayout" name="Slider2">
            <item>
             <widget class="QSlider" name="vsFOV">
              <property name="layoutDirection">
               <enum>Qt::LeftToRight</enum>
              </property>
              <property name="minimum">
               <number>45</number>
              </property>
              <property name="maximum">
               <number>180</number>
              </property>
              <property name="pageStep">
               <number>10</number>
              </property>
              <property name="value">
               <number>90</number>
              </property>
              <property name="sliderPosition">
               <number>90</number>
              </property>
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="invertedAppearance">
               <bool>false</bool>
              </property>
              <property name="invertedControls">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="LabelFOV">
              <property name="text">
               <string>Field of View</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="spFOV">
              <property name="minimum">
               <number>45</number>
              </property>
              <property name="maximum">
               <number>180</number>
              </property>
              <property name="value">
               <number>90</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="Slider1">
            <item>
             <widget class="QSlider" name="vsAngle">
              <property name="maximum">
               <number>360</number>
              </property>
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="LabelAngle">
              <property name="text">
               <string>Angle</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="spAngle">
              <property name="maximum">
               <number>360</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QFormLayout" name="flControlButtons">
          <item row="0" column="0">
           <widget class="QPushButton" name="fbReset">
            <property name="text">
             <string>Reset</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="fbInitializeTest">
            <property name="text">
             <string>Reset</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>10</y>
      <width>591</width>
      <height>581</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="Window">
     <property name="spacing">
      <number>0</number>
     </property>
     <item>
      <widget class="MyGLWidget" name="openGLWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>560</width>
         <height>480</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>560</width>
         <height>16777215</height>
        </size>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">myglwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>vsFOV</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spFOV</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>40</x>
     <y>410</y>
    </hint>
    <hint type="destinationlabel">
     <x>48</x>
     <y>460</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>vsAngle</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spAngle</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>126</x>
     <y>410</y>
    </hint>
    <hint type="destinationlabel">
     <x>167</x>
     <y>460</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
